src/loop/parser.ml

```diff
@@ -48,7 +48,7 @@ module Pipe
       | `Stdin ->
         let lang, file_loc, gen, _ = Logic.parse_input
             ?language:(State.input_lang st)
-            (`Stdin (Logic.Smtlib2 `Latest))
+            (`Stdin (Option.value ~default:(Logic.Smtlib2 `Latest) (State.input_lang st)))
```

Comment:
This should not be necessary in theory.
The `Logic.parse_input` function already takes the `?language` argument value over the ``` `Stdin l ```if it is present.

Reply:
Oh, I see, the problem is actually that this portion of code should bind the returned language from the `of_language` call, and return that instead of `l`, this should fix things.

For the read stdin but not in interactive mode: sure (and indeed in another PR/MR). I'll just have to think about how that interacts with the already present `mode` option (whether there needs to be another option or if a single option is enough/better).